WordPress/WordPressShare/UIImageView+Extensions.swift

```diff
@@ -0,0 +1,122 @@
+        let request = NSMutableURLRequest(URL: unwrappedUrl)
+        request.HTTPShouldHandleCookies = false
+        request.addValue("image/*", forHTTPHeaderField: "Accept")
+        
+        let scale = mainScreenScale
+        let session = NSURLSession.sharedSession()
+        let task = session.dataTaskWithRequest(request) { [weak self] data, response, error in
+            guard let data = data, let image = UIImage(data: data, scale: scale) else {
+                return
+            }
+            
+            dispatch_async(dispatch_get_main_queue()) {
+                // Update the Cache
+                Downloader.imagesCache[unwrappedUrl] = image
+                
+                // Refresh!
+                self?.image = image
+            }
+        }
+        
+        downloadTask = task
+        task.resume()
+    }
+    
+    
+    /// Downloads a resized Blavatar, meant to perfectly fit the UIImageView's Dimensions
+    ///
+    /// - Parameters:
+    ///     - url: The URL of the target blavatar
+    ///     - placeholderImage: Image to be displayed, temporarily, while the Download OP is executed
+    ///
+    public func downloadBlavatar(url: NSURL?, placeholderImage: UIImage?) {
+        guard let unwrappedURL = url else {
+            image = placeholderImage
+            return
+        }
+        
+        let components = NSURLComponents(URL: unwrappedURL, resolvingAgainstBaseURL: true)
+        components?.query = String(format: Downloader.blavatarResizeFormat, blavatarSize())
+        downloadImage(components?.URL, placeholderImage: placeholderImage)
+    }
+    
+    
+    /// Returns the desired Blavatar Side-Size
+    private func blavatarSize() -> Int {
+        var size = Downloader.defaultImageSize
+        
+        if !CGSizeEqualToSize(bounds.size, CGSizeZero) {
+            size = max(bounds.width, bounds.height);
+        }
+
+        size *= mainScreenScale
+    
+        return Int(size)
+    }
+    
+    /// Returns the Main Screen Scale
+    private var mainScreenScale : CGFloat {
+        return UIScreen.mainScreen().scale
+    }
+    
+    
+    /// Stores the current DataTask, in charge of downloading the remote Image
+    private var downloadTask : NSURLSessionDataTask? {
+        get {
+            return objc_getAssociatedObject(self, Downloader.taskKey) as? NSURLSessionDataTask
+        }
+        set {
+            let policy = objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC
+            objc_setAssociatedObject(self, Downloader.taskKey, newValue, policy)
+        }
+    }
+    
+    
+    /// Private helper structure
+    private struct Downloader
+    {
+        /// Default Blavatar Image Size
+        static let defaultImageSize = CGFloat(40)
+        
+        /// Blavatar Resize Query FormatString
+        static let blavatarResizeFormat = "d=404&s=%d"
+        
+        /// Stores all of the previously downloaded images
+        static var imagesCache = [NSURL : UIImage]()
```

Comment:
NSCache is probably a better option than a static dictionary. I'd say it's not a big deal since they are smallish images and the extension is short lived anyway, but since this file is an extension to `UIImageView` this could be problematic if it ends up in the main app.